R/draws.R

```diff
@@ -1,10 +1,112 @@
+#' base imputation model. The goal is to estimate the parameters of interest needed
+#' for the imputation phase (i.e. the regression coefficients and the covariance matrices
+#' from a MMRM model).
+#'
+#' The function distinguishes between the following methods:
+#' - Bayesian MI based on MCMC sampling: `draws` returns the draws
+#' from the posterior distribution of the parameters using a Bayesian approach based on
+#' MCMC sampling. This method can be adopted by specifying `method` using [method_bayes()].
+#' - Approximate Bayesian MI based on bootstrap: `draws` returns
+#' the draws from the posterior distribution of the parameters using a Bayesian approach,
+#' where the sampling from the posterior distribution is simulated fitting the MMRM model
+#' on bootstrap samples of the original dataset. This method can be adopted by
+#' specifying `method` through [method_approxbayes()].
+#' - Conditional mean imputation with bootstrap re-sampling: `draws` returns the
+#' ML/REML parameter estimates from the original dataset and from `n_samples` bootstrap samples.
+#' This method can be adopted by specifying `method` through [method_condmean()] with
+#' argument `type = "bootstrap"`.
+#' - Conditional mean imputation with Jackknife re-sampling: `draws` returns the
+#' ML/REML parameter estimates from the original dataset and from the leave-one-out samples.
+#' This method can be adopted by specifying `method` through [method_condmean()] with
+#' argument `type = "jackknife"`.
+#'
+#' The argument `data` must have one row per visit per subject. This means that incomplete
+#' outcome data must be set as `NA` instead of having the related row missing. Missing values
+#' in the covariates are not allowed. If `data` is incomplete
+#' then the [expand_locf()] helper function can be used to add in any missing rows, using
+#' Last Observation Carried Forward (LOCF) imputation to impute the covariates values.
+#'
+#' The argument `data_ice` contains information about the occurrence of ICEs. It is a
+#' `data.frame` with 3 columns:
+#' - A vector of characters related to the ids of the subjects that experienced
+#'   the ICE (must be named as `vars$subjid`).
+#' - A vector of levels of `data[[vars$visit]]` including the first visit after the ICE (must be named as `vars$visit`). If
+#'   multiple ICEs happen for the same subject, then only the first non-MAR visit should be set.
+#' - A vector of characters related to the imputation strategy to address the ICE (must be named as `vars$strategy`).
+#'   Possible imputation strategies are:
+#'   - `"MAR"`: Missing At Random.
+#'   - `"JTR"`: Jump To Reference.
+#'   - `"CIR"`: Copy Increments in Reference.
+#'   - `"CR"`: Copy Reference.
+#'   - `"JR"`: Jump to Reference.
+#'   - `"LMCF"`: Last Mean Carried Forward.
+#'   - User-defined imputation strategies can also be set.
+#'
+#' This argument is necessary at this stage since the model will be fit only on those observed data
```

Comment:
I am thinking that this might be easier to explain from the inverse  i.e.  

> The model is fitted onto the data after removing observation that occur after a non-MAR ICE.

Reply:
Agreed. However "non-MAR ICE" is not well said, I should say something like "The model is fitted onto the data after removing the observations addressed with an imputation strategy different from MAR"

Reply:
ok, let me try with this:

"This argument is necessary at this stage since the model is fitted after removing the observations whose underlying assumption is different from MAR. Thus, any observed data after `data_ice[[vars$visit]]` that is addressed with an imputation strategy different from MAR is discarded for the model fit. However such observations will not be discarded from the data in the imputation phase (performed with the function ([impute()]). To summarize, **at this stage only pre-ICE data and MAR post-ICE data are used**."

Reply:
I think I like that, a few minor tweaks:
discarded -> excluded
 MAR post-ICE data  ->  post-ICE data that is MAR

Reply:
ok!